execution/engine.go

```diff
@@ -273,6 +274,19 @@ func (e *Engine) IsEligibleForProposerBonus(marketID string, value *num.Uint) bo
 	return value.ToDecimal().GreaterThan(quantum.Mul(e.npv.minLpStakeQuantumMultiple))
 }
 
+// MarketsExpectKey checks if the currently existing markets use the provided key in their subscriptions.
+func (e *Engine) MarketsExpectKey(key string) bool {
+	for _, mkt := range e.markets {
+		for k := range mkt.mkt.TradableInstrument.Instrument.GetFuture().OraclesPubKeys() {
+			if strings.Compare(k, key) {
```

Comment:
I'm probably missing something but whats wrong with `if k == key`?

Reply:
I have added the logic here https://github.com/vegaprotocol/vega/pull/5534/commits/8ed6a685a32b5cdaf527f4bd083f72c17ff2a55a#diff-175c28f9d4bf1098cfc45b2c0071f08ea70dc56c117b273fee2ef110a876487eR15 it to aim for a more efficient comparison:  fail faster or/and in the first not matching rune, instead of traversing the whole strings.